runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java

```diff
@@ -85,7 +85,14 @@ public JavaStreamingContext call() throws Exception {
     EvaluationContext ctxt = new EvaluationContext(jsc, pipeline, options, jssc);
     // update cache candidates
     SparkRunner.updateCacheCandidates(pipeline, translator, ctxt);
-    pipeline.traverseTopologically(new SparkRunner.Evaluator(translator, ctxt));
+    try {
+      pipeline.traverseTopologically(new SparkRunner.Evaluator(translator, ctxt));
+    } catch (RuntimeException e) {
+      jssc.stop(false, false);
```

Comment:
Identifiers like `jsc` and `jssc` are very unreadable to me who is not doing Spark stuff every day.

Reply:
I created a separate issue/PR for this: https://github.com/apache/beam/pull/23204